packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncherMenu.tsx

```diff
@@ -1,61 +1,59 @@
-import React from 'react';
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
 import styles from '@patternfly/react-styles/css/components/AppLauncher/app-launcher';
 import { css } from '@patternfly/react-styles';
-import PropTypes from 'prop-types';
 import { ApplicationLauncherPosition } from './applicationLauncherConstants';
 import { DropdownArrowContext } from '../Dropdown/dropdownConstants';
-import ReactDOM from 'react-dom';
 import { keyHandler } from '../../helpers/util';
 import { KEY_CODES, KEYHANDLER_DIRECTION } from '../../helpers/constants';
 
-const propTypes = {
-  /** Anything which can be rendered as dropdown items */
-  children: PropTypes.node,
-  /** Classess applied to root element of dropdown menu */
-  className: PropTypes.string,
-  /** internal index of the item */
-  index: PropTypes.number,
-  /** Flag to indicate if menu is opened */
-  isOpen: PropTypes.bool,
-  /** Flag to indicate if menu should be opened on enter */
-  openedOnEnter: PropTypes.bool,
-  /** Indicates where menu will be alligned horizontally */
-  position: PropTypes.oneOf(Object.values(ApplicationLauncherPosition)),
-  /** Additional props are spread to the container component */
-  '': PropTypes.any // eslint-disable-line react/require-default-props
-};
+export interface ApplicationLauncherMenuProps extends React.HTMLProps<HTMLUListElement> {
+   /** Anything which can be rendered as dropdown items */
+   children?: React.ReactNode; 
+   /** Classess applied to root element of dropdown menu */
+   className?: string; 
+   /** internal index of the item */
+   index?: number; 
```

Comment:
I would keep the non optional ones as non optional ones (except for children and classname).

Reply:
I'm confused. I believe they were originally optional? I thought they were only required if the .isRequired is on the end for the js proptypes?

Reply:
yeah it's correct.